aiida_vasp/parsers/vasp.py

```diff
@@ -189,51 +189,48 @@ def parse(self, **kwargs):  # pylint: disable=too-many-return-statements
 
         # All quantities has been parsed, but there exit_codes reported from the parser
         # in this case, we return the code with the lowest status (hopefully the most severe)
-        if self._file_parser_exit_codes:
-            exit_codes = list(self._file_parser_exit_codes.values())
+        if self._object_parser_exit_codes:
+            exit_codes = list(self._object_parser_exit_codes.values())
             exit_codes.sort(key=lambda x: x.status)
             return exit_codes[0]
 
         return self.exit_codes.NO_ERROR
 
     def _parse_quantities(self):
         """
-        This method dispatch the parsing to file parsers
+        This method dispatch the parsing to object parsers
 
         :returns: A tuple of parsed quantities dictionary and a list of quantities failed to obtain due to exceptions
         """
         parsed_quantities = {}
-        # A dictionary for catching instantiated file parser objects
-        file_parser_instances = {}
+        # A dictionary for catching instantiated object parser objects
+        object_parser_instances = {}
         failed_to_parse_quantities = []
         for quantity_key in self._parsable_quantities.quantity_keys_to_parse:
```

Comment:
Here we will have to enter multiple context managers for each of the parsed files. so

```
parser = object_parser_cls(settings=self._settings, exit_codes=self.exit_codes, handler=self._get_handler(name))
```
should be fenced within a context:
```
with self._get_handler(name) as handler:
    parser = object_parser_cls(settings=self._settings, exit_codes=self.exit_codes, handler=handler)
```

This assumes that all file reading is done when the parser object is instantiated. Need to check if that is indeed the case for all file parsers.

Reply:
Question is also should we fence deep or high? I would think it would be good practice to let the fencing have as little code as possible in it and thus the fencing should be as close to the object processing as possible. Agreed? If so, that means in fact in `parsevasp`.

Reply:
No, that does not work. Okey, for our case it has to be high up as we fetch the handlers from the AiiDA data structures etc.

Reply:
Yes, we have to do it here, I think. A closed handler cannot be reopened. The file parsers should just read from the handler and avoid messing around with them.